Systems ref jcayouette (Content Lifecycle Update) (#433)
SUSE/doc-susemanager: modules/reference/pages/configuration/config-channels.adoc

[[ref.webui.config.channels]]
= Configuration Channels

> As mentioned above, {productname} manages both central and local configuration channels and files.

Reviewer: ```suggestion
{productname} manages both central and local configuration channels and files.
```
There's no guarantee that people have read the above, or that the content referenced will remain 'above'

> Central configuration management allows you to deploy configuration files to multiple systems (both traditional clients ([Management]) and Salt minions ([Salt])).

Reviewer: ```suggestion
Central configuration management allows you to deploy configuration files to multiple systems, and is available for both traditional and salt clients.
For traditional clients, use local configuration management.
For salt clients, use state  channels.
```

> Local configuration management is available for traditional clients ([Management]) only and allows you to specify overrides or configuration files that are not changed by subscribing the system to a central channel.

Reviewer: ```suggestion
For traditional clients, local configuration management is also available.
Local configuration management allows you to specify overrides, and select configuration files that are not changed when the system is subscribed to a central channel.
```

> A "`state channel`" is a type of a configuration channel but for Salt minion only.

Reviewer: ```suggestion
A state channel is a type of a configuration channel used only for Salt minions.
```

> For a state channel, the [path]``init.sls`` file is not auto-generated, the user creates and edits it.

Reviewer: ```suggestion
In state channels, the [path]``init.sls`` file is not automatically generated, you must manually create and edit it.
```

> Additionally, state channels can contain arbitrary configuration files that could be referenced from within the [path]``init.sls`` file.

Reviewer: ```suggestion
State channels can contain arbitrary configuration files that you can reference from within the [path]``init.sls`` file.
```

> Therefore, state channels effectively replace custom states.

Reviewer: ```suggestion

```

Added lines 6-7, which makes this distinction clear first up, and therefore makes this statement redundant.

> .Referencing Configuration Files with Organization ID

Reviewer: This should be a proper procedure, including steps below.

> [NOTE]

Reviewer: Move this admonition to the step it directly relates to, rather than at the top of the procedure.

====
You must reference configuration files with the `salt://` prefix, the organization ID, and the channel name.
For example, to reference [path]``/etc/motd`` use:

----
file.managed:  - source: salt://manager_org_1/`channel_name`/etc/motd
----
====

Central configuration or state channels must be created via the links on this page.

Click the name of the configuration channel to see the details page for that channel.
If you click the number of files in the channel, you are taken to the [guimenu]``List/Remove Files`` page of that channel.

> If you click the number of systems subscribed to the configuration channel, you are taken to the menu:Configuration Channel Details[Systems > Subscribed   Systems] page for that channel.

Reviewer: Remove the 'if you' statements, make it clear where these actions are taking place, and what the end goal is.

To create a new central configuration channel:

[[proc.config.channels.create.cfgch]]
.Procedure: Creating Central Configuration Channel

> . Click the [guimenu]``Create Config Channel`` link in the upper right corner of this screen.

Reviewer: ```suggestion
. In the {productname} {webui}, navigate to menu:Content Lifecycle Management[Projects], and click btn:[Create Config Channel].
```

. Enter a name for the channel.
. Enter a label for the channel. This field must contain only letters, numbers, hyphens (``-``) and underscores (``_``).

> . Enter a mandatory description for the channel that allows you to distinguish it from other channels. No character restrictions apply.

Reviewer: ```suggestion
. Enter a mandatory description for the channel that allows you to distinguish it from other channels.
```

. Click the btn:[Create Config Channel] button to create the new channel.

> . The following page is a subset of the [guimenu]``Channel Details`` page and has three tabs: [guimenu]``Overview`` , [guimenu]``Add Files`` , and [guimenu]``Systems`` . The [guimenu]``Channel Details`` page is discussed in <<config-config-channels-channel-details>>.

Reviewer: This could use a little tidy up. Name the page and describe what each thing actually is.

To create a new state channel with an [path]``init.sls`` file:

[[proc.config.channels.create.statech]]
.Procedure: Creating State Channel [Salt]

> . Click the [guimenu]``Create State Channel`` link in the upper right corner of this screen.

Reviewer: ```suggestion
. In the {productname} {webui}, navigate to menu:Content Lifecycle Management[Projects], and click btn:[Create State Channel].
```

. Enter a name for the channel.
. Enter a label for the channel. This field must contain only letters, numbers, hyphens (``-``) and underscores (``_``).

> . Enter a mandatory description for the channel that allows you to distinguish it from other channels. No character restrictions apply.

Reviewer: ```suggestion
. Enter a mandatory description for the channel that allows you to distinguish it from other channels.
```

. Enter the [guimenu]``SLS Contents`` for the [path]``init.sls`` file.
. Click the btn:[Create Config Channel] button to create the new channel.

> . The following page is a subset of the [guimenu]``Channel Details`` page and has four tabs: [guimenu]``Overview``, [guimenu]``List/Remove Files``, [guimenu]``Add Files``, and [guimenu]``Systems``. The [guimenu]``Channel Details`` page is discussed in <<config-config-channels-channel-details>>.

Reviewer: As per L40

[[config-config-channels-channel-details]]
== Configuration Channel Details

Overview::
The [guimenu]``Overview`` page of the [guimenu]``Configuration Channel Details``
page is divided into several panels.

Channel Information:::

> The panel provides status information for the contents of the channel.

Reviewer: ```suggestion
Provides status information for the contents of the channel.
```

Configuration Actions:::

> The panel provides access to the most common configuration tasks.

Reviewer: ```suggestion
Provides access to the most common configuration tasks.
```

For Salt minions, there is a link to edit the [path]``init.sls`` file.

Channel Properties [Management]:::

> By clicking the [guimenu]``Edit Properties`` link, you can edit the name, label, and description of the channel.

Reviewer: ```suggestion
Edit the name, label, and description of the channel by clicking the btn:[Edit Properties] button.
```

The [guimenu]``Filename/Path`` field is the absolute path where the file will be deployed.
+
You can set the [guimenu]``Ownership`` via the [guimenu]``user name`` and [guimenu]``group name`` and the [guimenu]``Permissions`` of the file when it is deployed.
+
If the client has SELinux enabled, you can configure [guimenu]``SELinux contexts`` to enable the required file attributes (such as user, role, and file type).
+
If the configuration file includes a macro (a variable in a configuration file), enter the symbol that marks the beginning and end of the macro.
For more information on using macros, see <<s3-sm-file-macros>>.
+
Import Files:::
To import files from other configuration channels, including any locally-managed channels, check the box to the left of any file you want to import.
Then click the btn:[Import Configuration File(s)] button.
+

NOTE: A sandbox icon (image:spacewalk-icon-sandbox.svg[Spacewalk Icon Sandbox,scaledwidth=0.9em]) indicates that the listed file is currently located in a local sandbox.
Files in a system's sandbox are considered experimental and could be unstable.
Use caution when selecting them for a central configuration channel.
+

Create File:::

> Create a configuration file, directory, or symbolic link from scratch to be included in the configuration channel.

Reviewer: ```suggestion
Create a configuration file, directory, or symbolic link to be included in the configuration channel.
```

> .Procedure: Creating a Configuration File, Directory, or Symbolic Link From Scratch

Reviewer: ```suggestion
.Procedure: Creating a Configuration File, Directory, or Symbolic Link
```

> ... Choose whether you want to create a text file, directory, or symbolic link in the [guimenu]``File Type`` section.

Reviewer: What do the three periods mean?
